common-docs/python/functions.md

```diff
@@ -0,0 +1,159 @@
+# Functions
+
+Functions are the fundamental building block of programs. Here is the simplest
+way to make a function that adds two numbers:
+
+```python
+// Named function
+def add(x: int, y: int):
+    return x + y
+
+sum = add(1, 2)
+```
+
+### ~ hint
+
+For the @boardname@, you must specify a [type](/python/types) for each function parameter. 
+
+### ~ 
+
+Functions can refer to variables outside of the function body.
+When they do so, they're said to `capture` these variables.
+
+```python
+z = 100
+
+def addToZ(x: int, y: int):
+    return x + y + z
+
+sum = addToZ(1, 2)
+```
+
+Let's add a return type to our add function:
+
+```python
+def add(x: int, y: int):
+    return x + y
+```
+
+Python can figure the return type out by looking at the return statements, so you can optionally leave this off in many cases.
+
+## Optional and Default Parameters
+
+In TypeScript, the number of arguments given to a function has to match the number of parameters the function expects.
```

Comment:
I wouldn't reference TypeScript here. The majority of students are likely learning Python as their first language. You can just cut out the whole TypesScript reference IMO